### ISSUE TITLE
Shadowing dependencies // Problem with scope 'implementation' vs. 'compile'

### ISSUE THREAD

**Issue by 1azyman** [JetBrains/gradle-intellij-plugin#656]
With gradle 7, `compile` dependency scope was deprecated/removed. Proper replacement should be `implementation` scope.
However dependencies added with `implementation` scope are added "after" `ideaIC`, which means all "shadowed" dependencies like apache velocity will not work correctly.

Sample repository: [example](https://github.com/1azyman/gradle7-scopes)

Dependency lists are in scope*.txt files

Changing scope `implementation` to `compile` will fix dependency issues and build ends with success.

**Comment by andysenn** [comment]
I also ran into this issue in an IntelliJ Platform Plugin project when I tried to use the `org.jetbrains.kotlin.plugin.serialization` Gradle plugin.

Despite explicitly adding a dependency of `org.jetbrains.kotlinx:kotlinx-serialization-json:1.3.2`, I would always get the following error at compile time:

> Your current kotlinx.serialization core version is too low, while current Kotlin compiler plugin 1.6.10 requires at least 1.0-M1-SNAPSHOT. Please update your kotlinx.serialization runtime dependency.

I was able to get this working by using the workaround @Pushpavel provided, though I'm not sure if this is a good long-term solution.

**Comment by hsz** [comment]
With the next release, the `ResolutionStrategy.SortOrder.DEPENDENCY_FIRST` resolution strategy will be enabled by default for `compileClasspath` and `testCompileClasspath` configurations.

It'll be possible to disable it, in terms of any potential troubles, with the build feature flag:

```properties
org.jetbrains.intellij.buildFeature.useDependencyFirstResolutionStrategy = false
```

**Comment by hsz** [comment]
Fix delivered with 1.7.0 isn't correct. Another one delivered with #1060 – see comments for more details.

**Comment by cjbrooks12** [comment]
I'm still getting this issue with plugin version v1.10.0. The above `DEPENDENCY_FIRST` workaround doesn't fix it, and the comments in #1060 suggest this issue has been fixed in 1.8.0, but it seems like it's still an issue. [Here's the configuration](https://github.com/copper-leaf/ballast/blob/dev/ballast-idea-plugin/build.gradle.kts) I'm using, v1.5.2 is the latest version I'm able to work with, anything later fails with the kotlinx.serialization version issue
